### PR TITLE
feat(cli): add batch URL download from input file (issue #136)

### LINKED ISSUE #136: feat: batch URL download from input file
## Summary

Add support for downloading multiple URLs from an input file, similar to aria2's `-i` / `--input-file` option.

## Usage

```bash
# Download all URLs from file
warpdl download -i urls.txt

# With options applied to all
warpdl download -i urls.txt --speed-limit 1MB --max-connection 8

# Mixed with direct URL
warpdl download -i urls.txt https://example.com/another.zip
```

### Simple (one URL per line)
```
https://example.com/file1.zip
https://example.com/file2.zip
# Comments start with #
https://example.com/file3.zip
```

### Extended (with per-file options)
```
https://example.com/file1.zip
  out=custom-name.zip
  dir=/custom/path
https://example.com/file2.zip
  header=Authorization: Bearer token123
```

### P0 - MVP
- [ ] `-i, --input-file` flag accepting file path
- [ ] One URL per line format
- [ ] Skip empty lines and comments (`#`)
- [ ] Queue all URLs (respects `--max-concurrent` from #135)
- [ ] Continue on individual download failures
- [ ] Summary at end: N succeeded, M failed

### P1 - Enhanced
- [ ] Extended format with per-file options (out, dir, header)
- [ ] `-` for stdin input (pipe URLs)
- [ ] JSON input format option
- [ ] `--input-format txt|json` flag

## Implementation

```go
// cmd/download.go
func parseInputFile(path string) ([]DownloadRequest, error) {
    file, err := os.Open(path)
    if err != nil {
        return nil, err
    }
    defer file.Close()
    
    var requests []DownloadRequest
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        line := strings.TrimSpace(scanner.Text())
        if line == "" || strings.HasPrefix(line, "#") {
            continue
        }
        requests = append(requests, DownloadRequest{URL: line})
    }
    return requests, scanner.Err()
}
```

## Related

- Depends on #135 (queue manager) for proper concurrent handling
- Useful for #132 (browser extension batch downloads)

### PR DESCRIPTION
**Pull request by Divkix**
## Summary
- Add `-i, --input-file` flag to download command for batch URL processing
- Parse input files with one URL per line, skip empty lines and `#` comments
- Continue downloading on individual failures, report summary at end
- Full test coverage with edge cases (Windows line endings, Unicode, special URL chars)

## Changes
- **New files**: `cmd/input_file.go`, `cmd/download_batch.go` (parser and batch logic)
- **Tests**: 37 new tests covering parser and batch download flows
- **Help text**: Updated download command examples showing input file usage

## Test plan
- [x] Input file parsing tests (empty, comments, whitespace, Unicode)
- [x] Batch download tests (all succeed, some fail, all fail, mixed)
- [x] Edge case tests (Windows CRLF, long URLs, special characters)
- [x] Integration tests with mock client
- [x] Manual E2E test with daemon
- [x] `go test -race ./...` passes
- [x] Coverage >= 80% for new files

Fixes #136

### CODE REVIEW

File: cmd/download.go [hunk 1]
```diff
@@ -200,3 +216,81 @@ Max Connections`+"\t"+`: %d
...
+		},
+	}
+
+	// Collect direct URLs from arguments (skip first if it's "download" command name)
+	directURLs := ctx.Args()
+
+	fmt.Printf("Input file: %s\n", inputFile)
+	if len(directURLs) > 0 {
+		fmt.Printf("Additional URLs: %d\n", len(directURLs))
+	}
```
[the remark this answers is not in the document]
Comment (Divkix) [reply cmd/download.go:282]: Fixed in df764ac. Updated the comment to accurately describe the behavior - `ctx.Args()` in urfave/cli doesn't include the command name, so no skipping is needed.

File: cmd/config.go [hunk 2]
```diff
@@ -51,9 +51,18 @@ bandwidth
 
 Example:
         warpdl https://domain.com/file.zip
```
[the remark this answers is not in the document]
Comment (Divkix) [reply cmd/config.go:53]: Keeping as-is. The current format is consistent with other examples in the file (see batch download examples at lines 57-59 which also don't use 'or' separators). Two consecutive examples is standard help text formatting.

File: cmd/input_file_test.go [hunk 3]
```diff
@@ -0,0 +1,810 @@
...
+	// The URL with inline comment should be parsed as-is
+	if len(result.URLs) != 2 {
+		t.Errorf("expected 2 URLs, got %d: %v", len(result.URLs), result.URLs)
+	}
+
+	// First URL should include the inline comment (no stripping)
+	expectedFirst := "https://example.com/file1.zip # inline comment"
+	if result.URLs[0] != expectedFirst {
+		t.Errorf("expected first URL %q, got %q", expectedFirst, result.URLs[0])
+	}
```
[the remark this answers is not in the document]
Comment (Divkix) [reply cmd/input_file_test.go]: Fixed in df764ac. Clarified the test comments to explain this is an intentional design decision: the `#` character is valid in URLs (fragment identifier like `https://example.com/page#section`), so we cannot strip content after `#`. URLs with spaces (like inline comments) will fail validation at download time, which is the correct behavior.
